unsafe/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -380,7 +405,13 @@ public void putNewKey(
       bitset.set(pos);
 
       // If there's not enough space in the current page, allocate a new page:
-      if (currentDataPage == null || PAGE_SIZE_BYTES - pageCursor < requiredSize) {
+      if (currentDataPage == null || PAGE_SIZE_BYTES - pageCursor - 8 < requiredSize) {
+        if (currentDataPage != null) {
```

Comment:
The fact that we have a check for `currentDataPage == null` in the outer if loop and `currentDataPage != null` here implies that the outer loop's first or branch is useless, since the body of the loop will never be executed if it holds.